sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -95,7 +95,7 @@ def _format_url_section(template, **kwargs):
             return template.format(**kwargs)
```

Comment:
With these changes, the infinite loop error on resource name `{object.value}` is fixed, however, this will still raise an IndexError on a resource name containing `{}` and will raise a ValueError on resource names containing a single  unmatched `}` (or `{`). Is this expected behavior? It seems that creating some resources with these names are allowed in the azure portal, so is there not a way to query for these resources through the python sdk?

Reply:
After some discussion, we are leaning towards encoding the url rather than hacking the format method.

@ava-silver could you help to check if this PR works for you?

Reply:
This seems like a much better solution! This should solve the issue 👍🏻